vector/src/main/java/im/vector/app/AutoRageShaker.kt

```diff
@@ -0,0 +1,273 @@
+                e2eDetectedFlow.emit(target)
+            }
+        }
+    }
+
+    private fun sendRageShake(target: E2EMessageDetected) {
+        bugReporter.sendBugReport(
+                context = context,
+                reportType = ReportType.AUTO_UISI,
+                withDevicesLogs = true,
+                withCrashLogs = true,
+                withKeyRequestHistory = true,
+                withScreenshot = false,
+                theBugDescription = "UISI detected",
+                serverVersion = "",
+                canContact = false,
+                customFields = mapOf("auto-uisi" to buildString {
+                    append("\neventId: ${target.eventId}")
+                    append("\nroomId: ${target.roomId}")
+                    append("\nsenderKey: ${target.senderKey}")
+                    append("\nsource: ${target.source}")
+                    append("\ndeviceId: ${target.senderDeviceId}")
+                    append("\nuserId: ${target.senderUserId}")
+                    append("\nsessionId: ${target.sessionId}")
+                }),
+                listener = object : BugReporter.IMXBugReportListener {
+                    override fun onUploadCancelled() {
+                        synchronized(alreadyReportedUisi) {
+                            alreadyReportedUisi.remove(ReportInfo(target.roomId, target.sessionId))
+                        }
+                    }
+
+                    override fun onUploadFailed(reason: String?) {
+                        synchronized(alreadyReportedUisi) {
+                            alreadyReportedUisi.remove(ReportInfo(target.roomId, target.sessionId))
+                        }
+                    }
+
+                    override fun onProgress(progress: Int) {
+                    }
+
+                    override fun onUploadSucceed(reportUrl: String?) {
+                        // we need to send the toDevice message to the sender
+
+                        coroutineScope.launch {
+                            try {
+                                activeSessionHolder.getSafeActiveSession()?.sendToDevice(
+                                        eventType = AUTO_RS_REQUEST,
+                                        userId = target.senderUserId,
+                                        deviceId = target.senderDeviceId,
+                                        content = mapOf(
+                                                "event_id" to target.eventId,
+                                                "room_id" to target.roomId,
+                                                "session_id" to target.sessionId,
+                                                "device_id" to target.senderDeviceId,
+                                                "user_id" to target.senderUserId,
+                                                "sender_key" to target.senderKey,
+                                                "matching_issue" to reportUrl
```

Comment:
`matching_issue` is a confusing name for this field, I'd expect that to be a link to a github issue or something; how would you feel about something more like `recipient_rageshake`?

Reply:
Sorry, didn't realize the rageshake server actually does create a github issue and responds with that; I'd had that disabled in my tests, and assumed we were looking at the URL of the submitted rageshake.